ibis/backends/flink/__init__.py

```diff
@@ -318,27 +350,50 @@ def create_table(
 
         if obj is None and schema is None:
             raise exc.IbisError("The schema or obj parameter is required")
+        if obj is not None and schema is not None:
+            raise exc.IbisError(
+                "Parameters `schema` and `obj` cannot be provided together. "
+                "Exactly one of them must be provided."
+            )
 
         if overwrite:
             self.drop_table(name=name, catalog=catalog, database=database, force=True)
 
         if isinstance(obj, pa.Table):
             obj = obj.to_pandas()
+
         if isinstance(obj, pd.DataFrame):
             qualified_name = self._fully_qualified_name(name, database, catalog)
             table = self._table_env.from_pandas(obj)
             # in-memory data is created as views in `pyflink`
             # TODO(chloeh13q): alternatively, we can do CREATE TABLE and then INSERT
             # INTO ... VALUES to keep things consistent
             self._table_env.create_temporary_view(qualified_name, table)
+
         if isinstance(obj, ir.Table):
-            # TODO(chloeh13q): implement CREATE TABLE for expressions
-            raise NotImplementedError
+            # TODO (mehmet): Double check this with the reviewers!
+            if not self.list_tables(name):
+                return ibis.memtable(obj, schema=obj.schema())
```

Comment:
There isn't support for creating a `memtable` out of an existing ibis expression, so i don't think you want this here.